From: Marco Nenciarini <mnencia@debian.org>
Date: Sat, 5 Mar 2011 11:19:07 +0100
Subject: split-protocols

Updated: Mon, 25 Nov 2013 20:11:59 -0500
Forwarded: no (Debian-specific)

Set default protocols value as empty and enable each
protocol in its own configuration file
---
 doc/example-config/dovecot.conf | 4 ++--
 src/master/master-settings.c    | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/doc/example-config/dovecot.conf b/doc/example-config/dovecot.conf
index a99d8aa..c802011 100644
--- a/doc/example-config/dovecot.conf
+++ b/doc/example-config/dovecot.conf
@@ -20,8 +20,8 @@
 # options. The paths listed here are for configure --prefix=/usr
 # --sysconfdir=/etc --localstatedir=/var
 
-# Protocols we want to be serving.
-#protocols = imap pop3 lmtp submission
+# Enable installed protocols
+!include_try /usr/share/dovecot/protocols.d/*.protocol
 
 # A comma separated list of IPs or hosts where to listen in for connections. 
 # "*" listens in all IPv4 interfaces, "::" listens in all IPv6 interfaces.
diff --git a/src/master/master-settings.c b/src/master/master-settings.c
index ce07ebd..ba20115 100644
--- a/src/master/master-settings.c
+++ b/src/master/master-settings.c
@@ -206,7 +206,7 @@ static const struct master_settings master_default_settings = {
 	.state_dir = PKG_STATEDIR,
 	.libexec_dir = PKG_LIBEXECDIR,
 	.instance_name = PACKAGE,
-	.protocols = "imap pop3 lmtp",
+	.protocols = "",
 	.listen = "*, ::",
 	.ssl = "yes:no:required",
 	.default_internal_user = "dovecot",
